step-plans/step-plans-base-artefacts/src/main/java/step/artefacts/handlers/UserFriendlyJsonObject.java

```diff
@@ -53,11 +53,7 @@ private Object unwrapSimpleValue(JsonValue fieldValue) {
             simpleValue = null;
         } else if (fieldValue.getValueType() == ValueType.NUMBER) {
             JsonNumber numberValue = (JsonNumber) fieldValue;
-            if (numberValue.isIntegral()) {
-                simpleValue = numberValue.intValue();
-            } else {
-                simpleValue = numberValue.doubleValue();
-            }
+            simpleValue = numberValue.numberValue();
```

Comment:
Would keep the initial switch and only change the line 57 to "simpleValue = numberValue.longValue();"